Product/Production/Services/DocumentDataSubmissionCore/src/test/java/gov/hhs/fha/nhinc/docdatasubmission/audit/transform/DocDataSubmissionAuditTransformsTest.java

```diff
@@ -0,0 +1,233 @@
+            }
+
+            @Override
+            protected String getRemoteHostAddress(Properties webContextProeprties) {
+                if (webContextProeprties != null && !webContextProeprties.isEmpty() && webContextProeprties.
+                    getProperty(NhincConstants.REMOTE_HOST_ADDRESS) != null) {
+                    return webContextProeprties.getProperty(NhincConstants.REMOTE_HOST_ADDRESS);
+                }
+                return AuditTransformsConstants.ACTIVE_PARTICIPANT_UNKNOWN_IP_ADDRESS;
+            }
+
+            @Override
+            protected String getWebServiceUrlFromRemoteObject(NhinTargetSystemType target, String serviceName) {
+                return WS_REQUEST_URL;
+            }
+        };
+
+        RegisterDocumentSetRequestType request = createRegisterDocumentSetRequestType();
+
+        AssertionType assertion = createAssertion();
+        LogEventRequestType auditRequest = transforms.transformRequestToAuditMsg(request, assertion,
+            createNhinTarget(), NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION, NhincConstants.AUDIT_LOG_NHIN_INTERFACE,
+            Boolean.TRUE, null, NhincConstants.NHINC_XDR_SERVICE_NAME);
+        testGetEventIdentificationType(auditRequest, NhincConstants.NHINC_XDR_SERVICE_NAME, Boolean.TRUE);
+        testCreateActiveParticipantFromUser(auditRequest, Boolean.TRUE, assertion);
+        testGetActiveParticipantDestination(auditRequest, Boolean.TRUE, null, WS_REQUEST_URL, REMOTE_IP);
+        testAuditSourceIdentification(auditRequest.getAuditMessage().getAuditSourceIdentification(), assertion);
+        testGetActiveParticipantSource(auditRequest, Boolean.TRUE, null, LOCAL_IP);
+        assertParticipantObjectIdentification(auditRequest.getAuditMessage());
+        assertEquals("AuditMessage.Request direction mismatch", auditRequest.getDirection(),
+            NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION);
+        assertEquals("AuditMessage.Request ServiceName mismatch", auditRequest.getEventType(),
+            NhincConstants.NHINC_XDR_SERVICE_NAME);
+    }
+
+    @Test
+    public void transformResponseToAuditMsg() throws ConnectionManagerException, UnknownHostException {
+        Properties webContextProperties = new Properties();
+        webContextProperties.setProperty(NhincConstants.WEB_SERVICE_REQUEST_URL, WS_REQUEST_URL);
+        webContextProperties.setProperty(NhincConstants.REMOTE_HOST_ADDRESS, REMOTE_IP);
+        webContextProperties.setProperty(NhincConstants.INBOUND_REPLY_TO, NhincConstants.WSA_REPLY_TO);
+        DocDataSubmissionAuditTransforms transforms = new DocDataSubmissionAuditTransforms() {
+            @Override
+            protected String getLocalHostAddress() {
+                return LOCAL_IP;
+            }
+
+            @Override
+            protected String getRemoteHostAddress(Properties webContextProeprties) {
+                if (webContextProeprties != null && !webContextProeprties.isEmpty() && webContextProeprties.
```

Comment:
Whats with the complicated logic here? If we are attempting to mock this stuff and set up the remote IP on line 116, why not just return it?